eos-ostree-remotes-config

```diff
@@ -28,16 +28,119 @@
 # This might only needed temporarily until ostree and flatpak get
 # updated to do sane remote modification.
 #
+# While here, look for any remotes configuration files that may have
+# ended up in /etc/ostree/remotes.d or /etc/flatpak/remotes.d. If there
+# are any, merge them into /ostree/repo/config and delete them.
+#
 # https://phabricator.endlessm.com/T19077
 # https://github.com/ostreedev/ostree/issues/1134
+# https://phabricator.endlessm.com/T22258
+
+from argparse import ArgumentParser
+from collections import OrderedDict
+from configparser import ConfigParser, ParsingError
+import glob
+import os
+import sys
+
+
+REMOTE_CONFIG_OPT = 'add-remotes-config-dir'
+REPO_PATH = '/ostree/repo'
+REPO_CONFIG_PATH = os.path.join(REPO_PATH, 'config')
+REPO_OBJECTS_PATH = os.path.join(REPO_PATH, 'objects')
+OSTREE_REMOTES_PATH = '/etc/ostree/remotes.d'
+FLATPAK_REMOTES_PATH = '/etc/flatpak/remotes.d'
+
+
+def read_config_file(config, path):
+    """Read the file at path into config
+
+    Returns False if the file doesn't exist or could not be parsed.
+    """
+    try:
+        print('Parsing', path)
+        with open(path) as f:
+            config.read_file(f)
+        return True
+    except FileNotFoundError as err:
+        print('Warning:', path, 'does not exist', file=sys.stderr)
+        print(err, file=sys.stderr)
+        return False
+    except ParsingError as err:
+        print('Warning: could not parse', path, file=sys.stderr)
+        print(err, file=sys.stderr)
+        return False
+
+
+aparser = ArgumentParser(
+    description='Configure /ostree/repo for safe remote adding')
+aparser.add_argument('-n', '--dry-run', action='store_true',
+                     help='only show what would be done')
+args = aparser.parse_args()
+
 
 # Check that /ostree/repo looks like a real repo
-[ -d /ostree/repo ] || exit 0
-[ -f /ostree/repo/config ] || exit 0
-[ -d /ostree/repo/objects ] || exit 0
-
-# Only set the option if it hasn't been set yet
-CONFIG_OPT="core.add-remotes-config-dir"
-if ! ostree --repo=/ostree/repo config get "$CONFIG_OPT" &>/dev/null; then
-    ostree --repo=/ostree/repo config set "$CONFIG_OPT" false
-fi
+if (not os.path.isdir(REPO_PATH) or
+    not os.path.isfile(REPO_CONFIG_PATH) or
+    not os.path.isdir(REPO_OBJECTS_PATH)):
+    sys.exit(0)
+
+# Read repo config
+repo_config = ConfigParser()
+if not read_config_file(repo_config, REPO_CONFIG_PATH):
+    # This is bad...
+    print('Error: could not read', REPO_CONFIG_PATH, file=sys.stderr)
+    sys.exit(1)
+
+# Set the core.add-remotes-config-dir option and write back now in case
+# there are errors later
+if repo_config.has_option('core', REMOTE_CONFIG_OPT):
+    value = repo_config.get('core', REMOTE_CONFIG_OPT)
+    if value != 'false':
```

Comment:
I'd be inclined to rework the logic here so we just change this setting to false every time.

Reply:
Yeah, you're right. I kept this from my original script where I was trying to be respectful of an existing setting, but I think we always want `false` or things will be broken on Endless.